libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline/validations/fields.rs

```diff
@@ -228,101 +228,28 @@ pub(super) fn validate_native_type_arguments(field: ScalarFieldWalker<'_, '_>, c
     };
 }
 
-pub(super) fn validate_default(field: ScalarFieldWalker<'_, '_>, ctx: &mut Context<'_>) {
-    use chrono::{DateTime, FixedOffset};
+pub(super) fn validate_default_value(field: ScalarFieldWalker<'_, '_>, ctx: &mut Context<'_>) {
```

Comment:
Here also. If you have a thing you know doesn't need validation, just exit early.

```patch
diff --git a/libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline/validations/fields.rs b/libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline/validations/fields.rs
index 723a03bc87..93f633d8c1 100644
--- a/libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline/validations/fields.rs
+++ b/libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline/validations/fields.rs
@@ -230,22 +230,26 @@ pub(super) fn validate_native_type_arguments(field: ScalarFieldWalker<'_, '_>, c
 
 pub(super) fn validate_default_value(field: ScalarFieldWalker<'_, '_>, ctx: &mut Context<'_>) {
     let model_name = field.model().name();
-    let default_mapped_name = field.default_value().and_then(|d| d.mapped_name());
     let default_attribute = field.default_attribute();
 
+    let default_mapped_name = match field.default_value().and_then(|d| d.mapped_name()) {
+        Some(name) => name,
+        None => return,
+    };
+
     // Named defaults.
-    if default_mapped_name.is_some() && !ctx.connector.supports_named_default_values() {
+    if !ctx.connector.supports_named_default_values() {
         ctx.push_error(DatamodelError::new_attribute_validation_error(
             "You defined a database name for the default value of a field on the model. This is not supported by the provider.",
             "default",
             default_attribute.unwrap().span,
         ));
-    }
 
-    if default_mapped_name.is_some() {
-        validate_db_name(model_name, default_attribute.unwrap(), default_mapped_name, ctx, false);
+        return;
     }
 
+    validate_db_name(model_name, default_attribute.unwrap(), default_mapped_name, ctx, false);
+
     let default_value = field.default_value().map(|d| d.value());
     let scalar_type = field.scalar_type();
```

Reply:
I intentionally didn't early return here because I thought having a `map` argument shouldn't prevent validating the value of the `@default` attribute. WDYT?

Reply:
A good point.